packages/esm-patient-search-app/src/patient-search.resource.tsx

```diff
@@ -93,68 +138,81 @@ export function useRecentlyViewedPatients(showRecentlySearchedPatients: boolean
   );
-
-  return result;
 }
 
-export function useRESTPatients(
-  patientUuids: string[],
-  searching: boolean = true,
+/**
+ * A custom React hook for fetching patient data from a REST API based on a list of patient UUIDs.
+ *
+ * @param patientUuids - An array of patient UUIDs to fetch data for. If null, no data will be fetched.
+ * @param isSearching - A boolean flag to determine if the search should be performed. Defaults to true.
+ * @param resultsToFetch - The number of results to fetch at a time. Defaults to 10.
+ * @param customRepresentation - A string representing the custom representation of patient data to fetch. Defaults to a predefined value 'v'.
+ *
+ * @returns An object containing:
+ *   - data: An array of fetched patient data
+ *   - isLoading: A boolean indicating if the initial data is being loaded
+ *   - fetchError: Any error that occurred during fetching
+ *   - hasMore: A boolean indicating if there are more patients to load
+ *   - isValidating: A boolean indicating if new data is being loaded
+ *   - setPage: A function to load more data
+ *   - currentPage: The current page of results
+ *   - totalResults: The total number of patients to be fetched
+ */
+
+export function useRestPatients(
+  patientUuids: string[] | null,
+  isSearching: boolean = true,
   resultsToFetch: number = 10,
-  customRepresentation: string = v,
-): PatientSearchResponse {
-  const { t } = useTranslation();
-
-  const getPatientUrl = (index) => {
-    if (index < patientUuids.length) {
-      return `${restBaseUrl}/patient/${patientUuids[index]}?v=${customRepresentation}`;
-    } else {
-      return null;
-    }
-  };
-
-  const { data, isLoading, isValidating, setSize, error, size } = useSWRInfinite<
-    FetchResponse<{ results: Array<SearchedPatient>; links: Array<{ rel: 'prev' | 'next' }>; totalCount: number }>,
-    Error
-  >(searching ? getPatientUrl : null, openmrsFetch, {
-    initialSize: resultsToFetch < patientUuids.length ? resultsToFetch : patientUuids.length,
-  });
-
-  useEffect(() => {
-    if (error) {
-      showNotification({
-        kind: 'error',
-        title: t('fetchingPatientFailed', 'Fetching patient details failed'),
-        description: error.message,
-      });
-    }
-  }, [error, t]);
+  customRepresentation: string = patientSearchCustomRepresentation,
+) {
+  const getPatientUrl = useCallback(
+    (index: number) => {
+      if (patientUuids && index < patientUuids.length) {
+        return `${restBaseUrl}/patient/${patientUuids[index]}?v=${customRepresentation}`;
+      } else {
+        return null;
+      }
+    },
+    [patientUuids, customRepresentation],
+  );
+
+  const shouldFetch = isSearching && patientUuids !== null && patientUuids.length > 0;
 
-  const results = useMemo(
+  const { data, isLoading, isValidating, setSize, error, size } = useSWRInfinite<FetchResponse<SearchedPatient>, Error>(
+    shouldFetch ? getPatientUrl : null,
+    openmrsFetch,
+    {
+      keepPreviousData: true,
```

Comment:
[keepPreviousData docs](https://swr.vercel.app/docs/advanced/understanding.en-US#return-previous-data-for-better-ux)